core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java

```diff
@@ -74,24 +79,25 @@ public CompletableFuture<Void> sweepAsync(String workflowId) {
 
     public void sweep(String workflowId) {
         WorkflowModel workflow = null;
+        StopWatch watch = new StopWatch();
```

Comment:
Hi @rq-dbrady , Instead of using stopwatch, Can we try to acquire a lock here and if successful, then we do repair or come out of sweep logic?

Reply:
So currently it follows the logic:

1. Create stopwatch.
2. Set the workflow context.
3. AcquireLock (if fail return from sweep)
4. Gets workflow from store.
5. Run repair
6. Start Stop Watch.
7. Decide
8. Remove From DeciderQueue
9. Release lock and Stop StopWatch.

Are you proposing we Acquire the lock at the method start and then start the stop watch ? so something like

```
 public void sweep(String workflowId) {
WorkflowModel workflow = null;
StopWatch watch = new StopWatch();
 if (!executionLockService.acquireLock(workflowId)) {
watch.start();
  //Do repair logic 
// Decide etc ...
}
```

Reply:
Hi @rq-dbrady , Yes, the first thing we should try is to acquire a lock. then start a stop watch, run decide, stop stopwatch. release a lock. Optionally we dont need stop watch to measure execution time. We can use `System.currentTimeMillis()`

Reply:
@manan164  , Applied changes in the logic order suggested  ! 

1. Acquire lock 
2. Run repair
3. Start timer
4. Decide
5. End timer
6. Release lock